src/components/ScrollComponent/ScrollComponent.jsx

```diff
@@ -0,0 +1,20 @@
+import { useRef, useCallback } from "react";
+
+export const ScrollComponent = (setPageNumber, totalCount, hasMore, loading) => {
+
+  const observer = useRef();
+  const lastBookElementRef = useCallback((node) => {
+    if(loading){
+      return 
+    }
+    if (observer.current) observer.current.disconnect();
+    observer.current = new IntersectionObserver((entries) => {
+ 
+      if (entries[0].isIntersecting && hasMore) {
+          setPageNumber((prevPage) => prevPage + totalCount);
+      }

```

Comment:
lets refactor this one to use children and has its own logic not passing around ref